drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -1808,20 +1894,42 @@ void RasterizerSceneGLES3::_render_geometry(RenderList::Element *e) {
 				if (state.debug_draw == VS::VIEWPORT_DEBUG_DRAW_WIREFRAME && s->array_wireframe_id) {
 
 					glDrawElementsInstanced(GL_LINES, s->index_wireframe_len, GL_UNSIGNED_INT, 0, amount);
+					if (storage->info.per_id_instance) {
+						RasterizerStorageGLES3::Info::Render render = {};
+						if (storage->info.rid_render_info_render.has(s->mesh)) {
+							render = storage->info.rid_render_info_render[s->mesh];
+						}
+						render.vertices_count += s->index_array_len * amount;
+						storage->info.rid_render_info_render[s->mesh] = render;
+					}
 					storage->info.render.vertices_count += s->index_array_len * amount;
 				} else
 #endif
 
 						if (s->index_array_len > 0) {
 
 					glDrawElementsInstanced(gl_primitive[s->primitive], s->index_array_len, (s->array_len >= (1 << 16)) ? GL_UNSIGNED_INT : GL_UNSIGNED_SHORT, 0, amount);
-
+					if (storage->info.per_id_instance) {
+						RasterizerStorageGLES3::Info::Render render = {};
+						if (storage->info.rid_render_info_render.has(s->mesh)) {
+							render = storage->info.rid_render_info_render[s->mesh];
+						}
+						render.vertices_count += s->index_array_len * amount;
+						storage->info.rid_render_info_render[s->mesh] = render;
+					}
 					storage->info.render.vertices_count += s->index_array_len * amount;
 
 				} else {
 
 					glDrawArraysInstanced(gl_primitive[s->primitive], 0, s->array_len, amount);
-
+					if (storage->info.per_id_instance) {
+						RasterizerStorageGLES3::Info::Render render = {};
+						if (storage->info.rid_render_info_render.has(s->mesh)) {
+							render = storage->info.rid_render_info_render[s->mesh];
+						}
+						render.vertices_count += s->array_len * amount;
+						storage->info.rid_render_info_render[s->mesh] = render;
+					}
```

Comment:
For this kind of thing it looks like it may be better just to store the counts in a temporary variable without an if,
```
int stats_vertex_count = 0;
stats_vertex_count += s->array_len * amount;
```
then have one if statement at the end of the function ..
```
if (storing_debug_stuff)
 ...
```
Just incrementing the counts isn't going to be much of a hit for 3d, even when stats are not being used, in fact the if statements may be more expensive than just doing the counts...

This way the stats would be less intrusive and easier to maintain.